packages/valory/skills/twitter_scoring_abci/behaviours.py

```diff
@@ -230,14 +230,16 @@ def _get_id_to_usernames(self, user_data: List) -> Dict:
         """Process Twitter user data"""
         return {i["id"]: "@" + i["username"] for i in user_data}
 
-    def _get_wallet_to_ids(self, tweets: List) -> Dict[str, str]:
+    def _get_wallet_to_ids(self, tweets: List) -> Generator[None, None, Dict[str, str]]:
         """Process Twitter user data"""
         result = {}
         for tweet in tweets:
             match = re.search(ADDRESS_REGEX, tweet["text"])
-            if match and WHITEPAPER_LINK in tweet["text"] and TAGLINE in tweet["text"]:
-                wallet_address = Web3.toChecksumAddress(match.group())
-                result[wallet_address] = tweet["author_id"]
+            if match and TAGLINE in tweet["text"]:
+                valid_link = yield from self.check_whitepaper_link(tweet["text"])
+                if valid_link:
+                    wallet_address = Web3.toChecksumAddress(match.group())
+                    result[wallet_address] = tweet["author_id"]
```

Comment:
I don't quite like this conditional nesting. Is there a way to include the "yield from" on line 239 into the conditional on line 238?

Reply:
The simplest thing might be to extract 238-242 to a `check_tweet` method.